File: Client/src/types/message.ts
export type Message = {
  id: number
  senderId: number
  senderDisplayName: string
  senderImageUrl?: string | null
  recipientId: number
  recipientDisplayName: string
  recipientImageUrl?: string | null
  content: string
  dateRead?: string | null
  messageSent: string
  currentUserSender?: boolean
}

export type MessageCreate = {
  recipientId: number;
  content: string;
};

File: Server/Models/Message.cs
using Server.Helpers;

namespace Server.Models;

public class Message
{
    public int Id { get; set; }
    public string Content { get; set; } = null!;
    public DateTime? DateRead { get; set; }
    public DateTime MessageSent { get; set; }
    public bool SenderDeleted { get; set; }
    public bool RecipientDeleted { get; set; }

    public int SenderId { get; set; }
    public AppUser Sender { get; set; } = null!;
    public int RecipientId { get; set; }
    public AppUser Recipient { get; set; } = null!;
}